controllers/manager/gatewayvmconfiguration_controller.go

```diff
@@ -560,13 +560,62 @@ func (r *GatewayVMConfigurationReconciler) reconcileVMSSVM(
 		return "", fmt.Errorf("vmss vm(%s) has empty os profile", to.Val(vm.InstanceID))
 	}
 
+	forceUpdate := false
+	// check LatestModelApplied
+	if vm.Properties.LatestModelApplied != nil && !to.Val(vm.Properties.LatestModelApplied) {
+		forceUpdate = true
```

Comment:
Could we only update when there is a failure? 

The reason here is when a vm is added to LB backend pool, it would always be shown as latest model not applied why the model is actually same as VMSS. So this means more APIs would be invoked whenever you PUT VM NIC

Reply:
OK... I do see one more PUT operation because of this `LatestModelApplied`. But it finally achieved `LatestModelApplied:true` after the additional PUT operation.

Reply:
Removed this if block. And let's watch the e2e/runner failures to see if the other checks are enough.